src/main/java/org/ice4j/ice/CheckList.java

```diff
@@ -209,6 +209,25 @@ protected synchronized CandidatePair getNextOrdinaryPairToCheck()
         return highestPriorityPair; //return even if null
     }
 
+    /**
+     * Returns a pair which matches the specified local and remote candidates'
+     * transport addresses, if any, otherwise null.
+     */
+    public synchronized CandidatePair findPairMatching(LocalCandidate local, RemoteCandidate remote)
```

Comment:
Why synchronized?  And what about just implementing equals for `CandidatePair`?

Reply:
`CandidatePair` already has `equals`, which tests full equality on the candidates (not just their `TransportAddress`es.